### PR TITLE
Make analytics report update job scheduling more efficient

### PR DESCRIPTION
**Pull request by SpecLad**
### Motivation and context

Currently, `schedule_analytics_report_autoupdate_job` attempts to debounce job scheduling by examining existing jobs before scheduling a new one. Unfortunately, the `scheduler.get_jobs` function, which it uses for this purpose, scales poorly. Not only does it fetch IDs of all scheduled jobs (and not just ones related to the current object), but it then fetches information about every job, one by one. The current logic doesn't even need this information, but RQ Scheduler provides no method to get just the IDs.

Replace the current logic with a new lightweight approach that uses a custom Redis key to block scheduling of additional jobs.

### How has this been tested?

Manual testing.

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have created a changelog fragment 
- ~~[ ] I have updated the documentation accordingly~~
- ~~[ ] I have added tests to cover my changes~~
- ~~[ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
- ~~[ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat/apps/analytics_report/report/create.py [hunk 1]
```diff
@@ -68,18 +68,18 @@ def _get_queue(self):
...
 
-    def _make_initial_queue_job_id(self, obj) -> str:
-        return f"{self._make_queue_job_prefix(obj)}initial"
+    def _make_queue_job_id(self, obj, start_time: timezone.datetime) -> str:
+        return f"{self._make_queue_job_prefix(obj)}-{start_time.timestamp()}"
 
-    def _make_regular_queue_job_id(self, obj, start_time: timezone.datetime) -> str:
-        return f"{self._make_queue_job_prefix(obj)}{start_time.timestamp()}"
+    def _make_autoupdate_blocker_key(self, obj) -> str:
+        return f"cvat:analytics:autoupdate-blocker:{self._make_queue_job_prefix(obj)}"
```
Comment (klakhov) [inline cvat/apps/analytics_report/report/create.py:82]: I dont think we should put something named `prefix` to the end of the key, its better to be in the beggining

**Review by klakhov — COMMENTED** [review]
Generally, patch works for me.

File: cvat/apps/analytics_report/report/create.py (hunk 1 above)
Comment (SpecLad) [reply cvat/apps/analytics_report/report/create.py:82]: Yeah, that's fair enough. I think the format of the key itself is fine, but I can rename the function to something more semantically fitting.

Comment (SpecLad) [reply cvat/apps/analytics_report/report/create.py]: Renamed.

**Review by klakhov — APPROVED** [review]

**Pull request merged**